FatmaQunnies/Task_manager.py

```diff
@@ -0,0 +1,110 @@
+from task import Task
+from urgent_task import UrgentTask
+from utils import validate_date, get_valid_input, validate_status, validate_priority
+
+
+tasks = []
+ 
+
+def add_task():
+  title_input = input("Enter task title ").strip()
+  title = get_valid_input(title_input)
+  description_input = input("Enter task description ").strip()
+  description = get_valid_input(description_input)
+  date_str = input("Enter the  date ")
+  due_date =validate_date(date_str)
+  is_urgent_inputt = input("Is this task urgent? (yes/no): ").strip()
+  is_urgent_input = get_valid_input(is_urgent_inputt).strip().lower()
+  if is_urgent_input in ['yes', 'y','ye']:
+        priority_input = input("Enter the priority (High/Medium/Low)  ").strip()
+        priority = validate_priority(priority_input)
+        task = UrgentTask(title, description, due_date, priority)
+        tasks.append(task)
+  else:
+    task = Task(title, description, due_date)
+    tasks.append(task)
```

Comment:
`tasks.append(task)` called twice in this if-else block, maybe we can have only one ?